pkg/util/vsphere/vm/power_state.go

```diff
@@ -59,16 +60,16 @@ func (b PowerOpBehavior) String() string {
 	}
 }
 
-// ErrInvalidSetPowerStateResult is returned if the channel returned by
-// SetPowerState receives a value that is a type other than SetPowerStateResult
-// or error.
-type ErrInvalidSetPowerStateResult struct {
-	SetPowerStateResult any
+// ErrInvalidPowerOpResult is returned if the channel returned by
+// SetPowerState and/or Restart receives a value that is a type other than
+// PowerOpResult or error.
+type ErrInvalidPowerOpResult struct {
+	PowerOpResult any
 }
 
 // Error enables this type to be returned as a Golang error object.
-func (e ErrInvalidSetPowerStateResult) Error() string {
-	return fmt.Sprintf("invalid result: %q", e.SetPowerStateResult)
+func (e ErrInvalidPowerOpResult) Error() string {
+	return fmt.Sprintf("invalid result: %q", e.PowerOpResult)
```

Comment:
Am conflicted on the `PowerOpBehavior` naming convention, it specifically handles the behavior when the VM is not running i.e. Suspend/Power Off. It does not describe the intent correctly.

Something like `PowerDownOpBehavior`, just an example, might be closer to the intent of the enum.

Reply:
It's a superset of the existing VMODL type [`VirtualMachinePowerOpType`](https://vdc-repo.vmware.com/vmwb-repository/dcr-public/1ef6c336-7bef-477d-b9bb-caa1767d7e30/82521f49-9d9a-42b7-b19b-9e6cd9b30db1/vim.vm.DefaultPowerOpInfo.PowerOpType.html) to include the value `trySoft`.